src/py/pyodide/http.py

```diff
@@ -148,8 +150,34 @@ async def bytes(self) -> bytes:
         self._raise_if_failed()
         return (await self.buffer()).to_bytes()
 
+    async def unpack_archive(self, extract_dir=None, format=None):
+        """Treat the data as an archive and unpack it into target directory.
+
+        Assumes that the file is an archive in a format that shutil has an
+        unpacker for. The arguments extract_dir and format are passed directly
+        on to ``shutil.unpack_archive``.
+
+        Parameters
+        ----------
+        extract_dir : str
+            Directory to extract the archive into. If not
+            provided, the current working directory is used.
+
+        format : str
+            The archive format: one of “zip”, “tar”, “gztar”, “bztar”.
+            Or any other format registered with ``shutil.register_unpack_format()``. If not
+            provided, ``unpack_archive()`` will use the archive file name extension
+            and see if an unpacker was registered for that extension. In case
+            none is found, a ``ValueError`` is raised.
+        """
+        filename = self._url.rsplit("/", -1)[-1]
+        f = NamedTemporaryFile(suffix=filename)
+        f.write(await self.bytes())
+        shutil.unpack_archive(f.name, extract_dir, format)
```

Comment:
It's a shame there is no way to do this without tempfiles, but it looks indeed to be the case.

Related https://bugs.python.org/issue39846, if that was possible with that feature we could have used it in micropip.